### PR TITLE
Refactored GCE auth utils.

### LINKED ISSUE #552: GCL sink doesn't work once the GCE token expires
The token is not refreshed periodically and after some time it causes the following error:
```
E0915 09:00:25.457853       1 external.go:87] failed to sync data to sinks - encountered the following errors: request to "https:/v1beta3/projects/marine-bison-90112/logs/kubernetes.io%2Fevents/entries:write" failed with status "401 Unauthorized" and response: "{\n  \"error\": {\n    \"code\": 401,\n    \"message\": \"Request had invalid authentication credentials.\",\n    \"status\": \"UNAUTHENTICATED\"\n  }\n}\n"
```

### PR DESCRIPTION
**Pull request by piosz**
fixes #552

### CODE REVIEW

File: util/gce/auth.go [hunk 1]
```diff
@@ -34,9 +46,104 @@ type AuthToken struct {
...
+	WaitForToken() (string, error)
+}
+
+// AuthTokenProvider is thread-safe.
+type realAuthTokenProvider struct {
+	// Token to use for authentication.
+	token string
+
+	// Protects <token>.
+	mutex sync.RWMutex
```
Comment (vishh) [inline util/gce/auth.go]: nit: We can embed the mutex into the parent struct.

File: util/gce/auth.go [hunk 2]
```diff
@@ -50,18 +157,20 @@ func VerifyAuthScope(expectedScope string) error {
...
+		token: token,
 	}
+}
 
-	var token AuthToken
-	err = json.Unmarshal([]byte(rawToken), &token)
-	if err != nil {
-		return AuthToken{}, fmt.Errorf("failed to unmarshal service account token with output %q: %v", rawToken, err)
-	}
+type fakeAuthTokenProvider struct {
```
Comment (vishh) [inline util/gce/auth.go]: How about placing this in a separate file?

**Comment by vishh** [comment]
Thanks for the fix @piosz! LGTM

File: util/gce/auth.go (hunk 1 above)
Comment (piosz) [inline util/gce/auth.go]: I think it is implementation specific. There is no mutex in fakeAuthTokenProvider.

File: util/gce/auth.go (hunk 2 above)
Comment (piosz) [inline util/gce/auth.go]: done

File: util/gce/auth.go (hunk 1 above)
Comment (vishh) [inline util/gce/auth.go]: I meant 
```go
type realAuthTokenProvider struct {
    sync.RWMutex
    ...
}
```

Comment (piosz) [inline util/gce/auth.go]: Got it! Sorry that I haven't understood before. Done.

**Pull request merged**